examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,491 @@
+# while the EDUCATION one is expressed in $/hours/years of education.
+# Looking at the coefficient plot to extrapolate feature importance could be
+# misleading as some of them vary on a small scale (as UNION or SEX that are
+# either 0 or 1), while feature like AGE varies a lot more, several decades.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std(axis=0).plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# For the reasons explained above, multiplying the coefficients by the
+# standard deviation of the related feature would improve our understanding on
+# feature importance on the model.
+# In that way, we emphasize that the
+# greater the variance of a feature, the larger the weight of the corresponding
+# coefficient on the output, all else being equal.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_ *
+    X_train_preprocessed.std(axis=0),
+    columns=['Coefficient importance'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The plot above tells us about dependencies between a specific feature and
+# the target when all other features remain constant, i.e., conditional
+# dependencies. An increase of the AGE will induce a decrease
+# of the WAGE when all other features remain constant. On the contrary, an
+# increase of the EXPERIENCE will induce an increase of the WAGE when all
+# other features remain constant.
+#
+# Checking the variability of the coefficients
+# --------------------------------------------
+#
+# We can check the coefficient variability through cross-validation.
+# If coefficients vary in a significant way changing the input dataset
+# the robustness of the model is not guaranteed.
+
+from sklearn.model_selection import cross_validate
+from sklearn.model_selection import RepeatedKFold
+
+cv_model = cross_validate(
+    model, X, y, cv=RepeatedKFold(n_splits=5, n_repeats=5),
+    return_estimator=True, n_jobs=-1
+)
+coefs = pd.DataFrame(
+    [est.named_steps['transformedtargetregressor'].regressor_.coef_ *
+     X_train_preprocessed.std(axis=0)
+     for est in cv_model['estimator']],
+    columns=feature_names
+)
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefs, orient='h', color='k', alpha=0.5)
+sns.boxplot(data=coefs, orient='h', color='cyan', saturation=0.5)
+plt.axvline(x=0, color='.5')
+plt.title('Coefficient importance variability')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
```

Comment:
```suggestion
###############################################################################
# The problem of correlated variables
# -------------------------------------------------
```
I like it when people see hints to take home messages by looking at the table of content.